util/generate-efs-substack.py

```diff
@@ -0,0 +1,64 @@
+                                                 Type="String",
+                                                 Description="SecurityGroup for Mount Target"))
+SubnetId = t.add_parameter(Parameter("SubnetId",
+                                     Type="String",
+                                     Description="SubnetId for Mount Target"))
+CreateEFS = t.add_condition("CreateEFS",
+                            And(Not(Equals(Select(str(0), Ref(EFS_options)), "NONE")), Equals(Select(str(1), Ref(EFS_options)), "NONE")))
+CreateMT = t.add_condition("CreateMT",
+                           And(Not(Equals(Select(str(0), Ref(EFS_options)), "NONE")), Equals(Select(str(7), Ref(EFS_options)), "NONE")))
+UsePerformanceMode = t.add_condition("UsePerformanceMode",
+                                     Not(Equals(Select(str(2), Ref(EFS_options)), "NONE")))
+UseEFSEncryption = t.add_condition("UseEFSEncryption",
+                                   Equals(Select(str(5), Ref(EFS_options)), "true"))
+UseEFSKMSKey = t.add_condition("UseEFSKMSKey",
+                               And(Condition(UseEFSEncryption), Not(Equals(Select(str(3), Ref(EFS_options)), "NONE"))))
+UseThroughputMode = t.add_condition("UseThroughputMode",
+                                    Not(Equals(Select(str(6), Ref(EFS_options)), "NONE")))
+UseProvisioned = t.add_condition("UseProvisioned",
+                                 Equals(Select(str(6), Ref(EFS_options)), "provisioned"))
+UseProvisionedThroughput = t.add_condition("UseProvisionedThroughput",
+                                           And(Condition(UseProvisioned), Not(Equals(Select(str(4), Ref(EFS_options)), "NONE"))))
+
+fs = t.add_resource(FileSystem("EFSFS",
+                               PerformanceMode=If(UsePerformanceMode, Select(str(2), Ref(EFS_options)), NoValue),
+                               ProvisionedThroughputInMibps=If(UseProvisionedThroughput, Select(str(4), Ref(EFS_options)), NoValue),
+                               ThroughputMode=If(UseThroughputMode, Select(str(6), Ref(EFS_options)), NoValue),
+                               Encrypted=If(UseEFSEncryption, Select(str(5), Ref(EFS_options)), NoValue),
+                               KmsKeyId=If(UseEFSKMSKey, Select(str(3), Ref(EFS_options)), NoValue),
+                               Condition=CreateEFS
+                               ))
+
+mt = t.add_resource(MountTarget("EFSMT",
+                                FileSystemId=If(CreateEFS, Ref(fs), Select(str(1), Ref(EFS_options))),
+                                SecurityGroups=[Ref(ComputeSecurityGroup)],
+                                SubnetId=Ref(SubnetId),
+                                Condition=CreateMT
+                                ))
+
+t.add_output(Output("FileSystemId",
+                    Description= "ID of the FileSystem",
+                    Value=If(CreateEFS, Ref(fs), Select("1", Ref(EFS_options)))))
+
+jsonFilePath = "TargetPath"
```

Comment:
`TargetPath` seems a placeholder, is this correct?

Reply:
Yes, it's just a placeholder, I will add a comment to clarify that